sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -285,22 +312,24 @@ public virtual async Task<Response> UploadAsync<T>(string ruleId, string streamN
             Argument.AssertNotNullOrEmpty(logs, nameof(logs));
 
             // Calculate the number of threads to use.
-            // If there are 0 workers or an UploadLogsOptions object was not provided, method will run with 5 workers. Otherwise will run in parallel with number of workers given.
-            int _maxWorkerCount = (options == null || options.MaxConcurrency <= 0) ? DefaultParallelWorkerCount : options.MaxConcurrency;
+            int _maxWorkerCount = (options == null) ? new UploadLogsOptions().MaxConcurrency : options.MaxConcurrency;
```

Comment:
Why not just store the newly created options so we don't have to use check for nulls later on?